content/src/logic/database-queries/snapshots-queries.ts

```diff
@@ -33,3 +35,103 @@ export async function* streamActiveDeployments(
     }
   }
 }
+
+export async function* streamActiveDeploymentsInTimeRange(
+  components: Pick<AppComponents, 'database'>,
+  timeRange: TimeRange
+): AsyncIterable<DeploymentWithAuthChain> {
+  // IT IS IMPORTANT THAT THIS QUERY NEVER CHANGES. ORDER IS NOT GUARANTEED  `
+  for await (const row of components.database.streamQuery(
+    SQL`
+    SELECT
+      entity_id,
+      entity_type,
+      entity_pointers,
+      auth_chain,
+      date_part('epoch', local_timestamp) * 1000 AS local_timestamp
+    FROM deployments d
+    WHERE d.deleter_deployment IS NULL
+    AND to_timestamp(${timeRange.initTimestampSecs}) <= local_timestamp
+    AND local_timestamp <= to_timestamp(${timeRange.endTimestampSecs})
+    `,
+    { batchSize: 1000 },
+    'stream_active_deployments_in_timerange'
+  )) {
+    yield {
+      entityId: row.entity_id,
+      entityType: row.entity_type,
+      pointers: row.entity_pointers,
+      localTimestamp: row.local_timestamp,
+      authChain: row.auth_chain
+    }
+  }
+}
+
+export async function findSnapshotsStrictlyContainedInTimeRange(
+  components: Pick<AppComponents, 'database'>,
+  timerange: TimeRange
+): Promise<NewSnapshotMetadata[]> {
+  const query = SQL`
+  SELECT
+    hash,
+    date_part('epoch', init_timestamp)  AS "initTimestampSecs",
+    date_part('epoch', end_timestamp)  AS "endTimestampSecs",
+    replaced_hashes as replacedSnapshotHashes,
+    number_of_entities as numberOfEntities
+  FROM snapshots
+  WHERE to_timestamp(${timerange.initTimestampSecs}) <= init_timestamp
+  AND end_timestamp <= to_timestamp(${timerange.endTimestampSecs})
```

Comment:
https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-between/

Reply:
The committed version is inverted and shouldn't work heh that's why I suggested the BETWEEN operator to prevent this kind of typing errors

Reply:
Oh nevermind those are different fields